### DISCUSSION TITLE
Weird GET requests being logged

### DISCUSSION THREAD

**Discussion by ahmedabdou14** [fastapi/fastapi#10991] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python

```

### Description

I usually see weird GET requests in my local FastApi app. Today I saw many of them and had to ask if anyone is facing the same thing

![image](https://github.com/tiangolo/fastapi/assets/104530599/2ae2a8db-02ea-495c-a7a4-8b0dac4059ef)

### Here some info about the local env
The app is dockerized and is running behind Traefik reverse proxy.
Here are the dependencies:
```env
# App
uvicorn==0.25.0
fastapi==0.109.0
sse-starlette==1.8.2
pydantic==2.5.3
pydantic-settings==2.1.0
email-validator==2.1.0.post1
python-multipart==0.0.6
async-timeout==4.0.3
tenacity==8.2.3
annotated-types==0.6.0

# Email
sendgrid==6.11.0

# Security
python-jose[cryptography]==3.3.0
passlib[bcrypt]==1.7.4
argon2-cffi==23.1.0

# Database
asyncpg==0.29.0
alembic==1.13.1
SQLAlchemy==2.0.25

# Cache
fastapi-limiter==0.1.6
redis[hiredis]==5.0.1
rq==1.15.1

# Dev
pre-commit==3.6.0
mypy==1.8.0
ruff==0.1.12
bandit==1.7.6
interrogate==1.5.0
pytest==7.4.3
coverage==7.4.0
```

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.12

**Comment by jgould22** [comment]
Is your app exposed to the internet?

**Reply by ahmedabdou14** [reply]
I guess so yes, `sendgrid` is used to send emails
I believe this is the only dependency that uses the internet

**Reply by JavierSanchezCastro** [reply]
In my case I've deployed my FastAPI server dockerized in a Azure VM and also I receive many calls to endpoints I don't have, as I read it's normal since it's exposed to internet. Also (I'm not really sure about that) but I think the log of "Invalid HTTP request" comes when you don't have a SSL certificate running in uvicorn, and some client perform a HTTPS call.

**Reply by jgould22** [reply]
> I guess so yes, `sendgrid` is used to send emails
> I believe this is the only dependency that uses the internet

Sending connections out is different than receiving connections

If you app is accessible via public IP this is expected. The internet is a scary place and there are many things scanning the internet and checking and looking for known exploits or software that has known exploits.

**Reply by ahmedabdou14** [reply]
> In my case I've deployed my FastAPI server dockerized in a Azure VM and also I receive many calls to endpoints I don't have, as I read it's normal since it's exposed to internet. Also (I'm not really sure about that) but I think the log of "Invalid HTTP request" comes when you don't have a SSL certificate running in uvicorn, and some client perform a HTTPS call.

Yes i do not have an SSL certification for uvicorn as it is all local.
But i am not sure if this is the case tbh. What i initially thought is maybe `uvicorn` testing if it is connected to the internet but why would that be the case or why would it send all these random requests all of a sudden

**Reply by ahmedabdou14** [reply]
> > I guess so yes, `sendgrid` is used to send emails
> > I believe this is the only dependency that uses the internet
> 
> Sending connections out is different than receiving connections
> 
> If you app is accessible via public IP this is expected. The internet is a scary place and there are many things scanning the internet and checking and looking for known exploits or software that has known exploits.

True, but i dont think it is possible if it is a local environment, and no ports are exposed outside my home network either.
The requests are `GET`  requests and it is another indication that it might be something internal to any of `FastApi` dependencies maybe.

**Reply by jgould22** [reply, marked as answer]
I would make sure you are positive they are not exposed outside your network.

This log would be familiar to anyone who has hosted a web server that is exposed to the public internet.

You should try and bind your app strictly to the loopback address and see if you still observe this behaviour.

**Reply by ahmedabdou14** [reply]
It is 100% local. Yes i am running them behind Traefik reverse proxy, but all are under localhost (127.0.0.X) ) loopback address.
I setup Let'sEncrypt for Traefik for TLS but that doesn't even work because it is local. But those logs are coming from FastApi server and not Traefik so that excludes any Traefik blame.

**Reply by dolfinus** [reply]
That's definitely not related to FastAPI. This is some kind of scraper bot which tries to find vulnerability in your application by sending a bunch of requests which are known vulnerabilities in some CMSes like Wordpress

**Reply by ahmedabdou14** [reply]
Thank you @jgould22 
Somehow, my pc port was exposed to the internet as you said. I still can't recall when did i expose it in my home network as all ports are not exposed by default.

lesson learnt is, even if you are sure, check again
